lib/iris/cube.py

```diff
@@ -1332,6 +1332,10 @@ def __init__(
             for ancillary_variable, avdims in ancillary_variables_and_dims:
                 self.add_ancillary_variable(ancillary_variable, avdims)
 
+        # Set ordered_axis property - if we have more that one coord system then
+        # it should be True.
+        self._ordered_axes = len(self.coord_systems()) > 1
+
```

Comment:
This creates an inconsistency, see below. Given the potential for breaking changes, I think it would be best to always instantiate with `self._ordered_axes = False`.

To provide round-tripping when the extended grid mapping syntax is used, we should find a way to correctly set the property during loading, instead.

---

```python
my_cube = Cube(my_data)
my_cube.add_dim_coord(lon, 0)
my_cube.add_dim_coord(lat, 1)
my_cube.add_aux_coord(x, [0, 1])
my_cube.add_aux_coord(y, [0, 1])
print(my_cube.ordered_axes)
```

`False`

---

```python
my_cube = Cube(
    my_data,
    dim_coords_and_dims=[(lon, 0), (lat, 1)],
    aux_coords_and_dims=[(x, [0, 1]), (y, [0, 1])],
)
print(my_cube.ordered_axes)
```

`True`